### PR TITLE
Update terminology for lookup index -> token lookup index

### PR DESCRIPTION
**Pull request by Hunterness**
Does the update from https://github.com/neo-technology/neo4j-manual-modeling/pull/2065 in the cypher manual as well.

Includes some parts from https://github.com/neo4j/neo4j-documentation/pull/1090 (to hopefully avoid (or easier to fix) conflicts since I needed to update the same code in both).

### CODE REVIEW

File: cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc [hunk 1]
```diff
@@ -130,25 +130,25 @@ CREATE LOOKUP INDEX [index_name] [IF NOT EXISTS]
...
 
-Lookup indexes do not support any `OPTIONS` values.
+Token lookup indexes do not support any `OPTIONS` values.
 
 | [source, cypher, role=noplay]
 ----
 DROP INDEX index_name [IF EXISTS]
 ----
-| Drop an index, either a node, a relationship, or a lookup index.
+| Drop an index, either a btree, fulltext, or token lookup index.
```
Comment (burqen) [inline cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc]: I see a little inconsistency if we are using btree or b-tree here. In operations manual we use b-tree in text unless we are specifically referring to syntax or name in code such as `SHOW INDEXES#BTREE`, `org.neo4j.graphdb.schema.IndexType#BTREE`, `native-btree-1.0`
https://neo4j.com/docs/operations-manual/current/performance/index-configuration/

Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc]: that should probably be unified yes, I think we have a mix here

Comment (Hunterness) [inline cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc]:
```suggestion
| Drop an index, either a b-tree, fulltext, or token lookup index.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala [hunk 2]
```diff
@@ -297,8 +297,7 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
     profileQuery(
       title = "Create Index",
       text =
-        """The `CreateIndex` operator creates an index. This index can either be on a property for all nodes or relationships having a certain label or relationship type,
-          |or it can be a lookup index for all nodes or relationships having any label or relationship type.
+        """The `CreateIndex` operator creates an index. This index can either be a btree, fulltext, or token lookup index.
```
Comment (Hunterness) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]:
```suggestion
        """The `CreateIndex` operator creates an index. This index can either be a b-tree, fulltext, or token lookup index.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala [hunk 3]
```diff
@@ -251,7 +251,7 @@ class SchemaIndexTest extends DocumentingTestBase with QueryStatisticsTestSuppor
     prepareAndTestQuery(
       title = "Drop an index",
       text =
-        """An index can be dropped using the name with the `DROP INDEX index_name` command. This command can drop node, relationship or lookup indexes.
+        """An index can be dropped using the name with the `DROP INDEX index_name` command. This command can drop btree, fulltext, or token lookup indexes.
```
Comment (Hunterness) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala]:
```suggestion
        """An index can be dropped using the name with the `DROP INDEX index_name` command. This command can drop b-tree, fulltext, or token lookup indexes.
```

File: cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc (hunk 1 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc]: I think that the old code hopefully is consistent so for now I just updated the new places :)

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc [hunk 4]
```diff
@@ -51,7 +51,7 @@ This section comprises a glossary of all the keywords -- grouped by category and
 |<<query-return, RETURN ... [AS]>>                    | Projecting   |  Defines what to include in the query result set.
 |<<query-set, SET>>                          | Writing     |  Update labels on nodes and properties on nodes and relationships.
 |<<administration-constraints-list-constraint, SHOW [ALL\|UNIQUE\|NODE [PROPERTY] EXIST[ENCE]\|REL[ATIONSHIP] [PROPERTY] EXIST[ENCE]\|[PROPERTY] EXIST[ENCE]\|NODE KEY] CONSTRAINT[S]>> | Schema | List constraints in the database, either all or filtered on type. Also allows `WHERE` and `YIELD` clauses.
-|<<administration-indexes-list-indexes, SHOW [ALL\|BTREE\|FULLTEXT\|LOOKUP] INDEX[ES]>> | Schema | List indexes in the database, either all or filtered on B-tree, fulltext or lookup indexes. Also allows `WHERE` and `YIELD` clauses.
+|<<administration-indexes-list-indexes, SHOW [ALL\|BTREE\|FULLTEXT\|LOOKUP] INDEX[ES]>> | Schema | List indexes in the database, either all or filtered on B-tree, fulltext or token lookup indexes. Also allows `WHERE` and `YIELD` clauses.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/glossary.asciidoc]:
```suggestion
|<<administration-indexes-list-indexes, SHOW [ALL\|BTREE\|FULLTEXT\|LOOKUP] INDEX[ES]>> | Schema | List indexes in the database, either all or filtered on b-tree, fulltext or token lookup indexes. Also allows `WHERE` and `YIELD` clauses.
```

File: cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc [hunk 5]
```diff
@@ -159,7 +159,7 @@ SHOW [ALL\|BTREE\|FULLTEXT\|LOOKUP] INDEX[ES]
     [WHERE expression]
     [RETURN field[, ...] [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
 ----
-| List indexes in the database, either all or filtered on B-tree, fulltext, or lookup indexes.
+| List indexes in the database, either all or filtered on B-tree, fulltext, or token lookup indexes.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc]:
```suggestion
| List indexes in the database, either all or filtered on b-tree, fulltext, or token lookup indexes.
```

File: cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/IndexTest.scala [hunk 6]
```diff
@@ -124,15 +124,15 @@ Create a composite index on nodes with label `Person` and the properties `name`
 CREATE LOOKUP INDEX lookup_index_name FOR (n) ON EACH labels(n)
 ###
 
-Create a lookup index on nodes with any label with the name `lookup_index_name`.
+Create a token lookup index on nodes with any label with the name `lookup_index_name`.
```
Comment (AlexicaWright) [inline cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/IndexTest.scala]:
```suggestion
Create a token lookup index with the name `lookup_index_name` on nodes with any label .
```

**Review by AlexicaWright — COMMENTED** [review]
For consistency, b-tree should be b-tree everywhere (i.e. not btree or B-tree). 
Also, would it be worth mentioning that the token lookup index is present by default (in 4.3 databases)? Also, should there be a warning about dropping the node label lookup index?

**Comment by Hunterness** [comment]
> Also, would it be worth mentioning that the token lookup index is present by default (in 4.3 databases)? Also, should there be a warning about dropping the node label lookup index?

Might be a good idea yes

Edit: I don't know how to formulate a good note :(

**Review by AlexicaWright — APPROVED** [review]
LGTM 👍

**Comment by Hunterness** [comment]
Already cherry-picked as well: https://github.com/neo4j/neo4j-documentation/pull/1122

**Pull request merged**